app/src/main/java/de/danoeh/antennapod/fragment/preferences/AutoDownloadPreferencesFragment.java

```diff
@@ -175,10 +181,91 @@ private void buildEpisodeCleanupPreference() {
     }
 
     private void setSelectedNetworksEnabled(boolean b) {
+        if (permissionHelper.showPermissionRequestPromptOnAndroid10IfNeeded(b)) {
+            return;
+        }
+
         if (selectedNetworks != null) {
             for (Preference p : selectedNetworks) {
                 p.setEnabled(b);
             }
         }
     }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        permissionHelper.doOnRequestPermissionsResult(requestCode, permissions, grantResults);
+    }
+
+    private class PermissionHelper {
```

Comment:
Why is this another class? Can the class be moved to another file? Otherwise, I would prefer to just call the methods instead of adding a class.

Reply:
It's a matter of style. I find grouping Permission handling in another class makes (often verbose) permission handling logic separate from the main logic. It makes the codes easier to read / navigate.

Let me know what you think.

Reply:
The code that needs to be added to the methods is quite short. If you extract the code to another class, a reader has to scroll around and maybe get lost. All in all, I think that this PR is quite hard to read. You don't have to extract a method of one line of code that is called one single time. If you just write the line, one can directly read what it does.

Reply:
The latest check-in has removed the inner class, similar to what the rest of the AntennaPod codebase does.

Regarding the PR being quite hard to read, is the permission handling inner class the main reason? Or is it something else?

The permission handling inner class is really a scoping trick. I find it clearer, so that when someone wants to understand what the fragment does, he/she can easily focus on the main logic and filter out the permission logic (usually edge case). Or if he/she wants to reviews permission logic, he/she can just focus on the inner class without scrolling back and forth searching for relevant methods, fields, etc.

Granted the rest of the code base does not follow such pattern, so I am not too hung up on it either. I'm just surprised by the feedback the inner class making the codes harder to read.